frontend/readhistory.lua

```diff
@@ -258,7 +258,7 @@ function ReadHistory:updateItemByPath(old_path, new_path)
             self.hist[i].file = new_path
             self.hist[i].text = new_path:gsub(".*/", "")
             self:_flush()
-            self:reload()
+            self:reload(true)
```

Comment:
Not related to your changes, so not expecting a fix in this PR if need be.
But does this mean that if you select and move 20 files, we will reload the history 20 times ?

Reply:
Yes, it does. And the worst thing is that we will `flush()` 20 times.
This behaviour cannot be fixed with minor changes, full restructuring is required.